src/Chapter/DataStructures.elm

```diff
@@ -4,10 +4,120 @@ data =
   { title = "Data Structures"
   , content = """
 
+Up until this point, you've used lots of different values of lots of different types, but have only had one way of storing multiple values, namely: lists. A list is an example of an extremely simple data structure. In this section, we'll take a look at some of the other data structures available to you in elm.
+
 # Union Types
 
+<dfn>Union types</dfn> are essentially a way to join different types together, hence their name. They allow you to store many values of different types inside of a single data structure. For those of you with a java background; as the <a href="http://elm-lang.org/learn/Union-Types.elm">Elm website</a> puts it: "Think of [union types] as enums on steroids."
+
+Let's start off with defining a very simple union type:
+
+```elm
+type Animal
+  = Rabbit
+  | Dog
+  | Cat
+
+myBunny : Animal
+myBunny = Rabbit
+
+myPuppy : Animal
+myPuppy = Dog
+
+myKitty : Animal
+myKitty = Cat
+```
+<p class=note>You might have noticed that the definition of Animal is written on multiple lines after the first indented. This is allowed (and often encouraged!) in Elm. Doing this has no impact on the program, so the first statement could have been written as `type Animal = Rabbit | Dog | Cat`, but the code is much cleaner when the statement is split up across multiple lines.
+
+The first statement defines the type Animal such that something of the type Animal can have three possible values: Rabbit, Dog and Cat. In this case, the pipe (`|`) character is used to mean "or," in that an Animal can be a Rabbit **or** Dog **or** Cat. The rest of the code just creates a few Animals. As you can see from the type declarations, despite having a three different values, each still has the type Animal.
+
+Right now, it might seem like union types really are just enumerations, so let's now add in the data component. Let's say we want to keep some information about each different animal. For the rabbits, we need to know their color, but for dogs and cats we need to know their owner's name and the Animal's age. Let's redefine Animal so that it allows for this.
+
+```elm
+type Animal
+  = Rabbit String
+  | Dog String Int
+  | Cat String Int
+
+myBunny : Animal
+myBunny = Rabbit "white"
+
+cindysPuppy : Animal
+cindysPuppy = Dog "Cindy" 3
+
+johnsCat : Animal
+johnsCat = Cat "John" 2
+```
+<p class=note> You'll notice that instead of `number`, here we use `Int`.   `number` is actually a special type which can be either an `Int` (an integer value) or a `Float` (a decimal number). Elm normally decides for you which type `number` should be, but when defining union types, you need to specify which you'd like to use. In this case, we only want whole number ages, so we use `Int`.
+
+<p class=note>`Rabbit String`, `Dog String Int`, `Cat String Int` are all the different <dfn>type constructors</dfn> of Animal. They're called this because they are used to create or *construct* values of the type Animal.
```

Comment:
Technically `Rabbit`, `Dog`, and `Cat` are type constructors. `Rabbit String` is the syntax for declaring the function `Rabbit : String -> Animal`